stac_fastapi/api/stac_fastapi/api/routes.py

```diff
@@ -100,28 +101,38 @@ def add_route_dependencies(
     Allows a developer to add dependencies to a route after the route has been
     defined.
 
+    "*" can be used for path or method to match all allowed routes.
+
     Returns:
         None
     """
     for scope in scopes:
+        _scope = copy.deepcopy(scope)
         for route in routes:
-            match, _ = route.matches({"type": "http", **scope})
+            if scope["path"] == "*":
+                _scope["path"] = route.path
+
+            if scope["method"] == "*":
+                _scope["method"] = list(route.methods)[0]
```

Comment:
Is this correct or should we apply dependencies to each method individually?

Reply:
I believe so, the method is only used to get a full match with the route [here](https://github.com/stac-utils/stac-fastapi/pull/705/files#diff-3f814cfc24182540359cf16e8a3cedff38803d2feb6e489f7fc83b2e46758287R118-R120). The dependency is then applied to the route so is not specific to the method.